common/record_logger.hpp

```diff
@@ -0,0 +1,317 @@
+#pragma once
+
+#include <any>
+#include <atomic>
+#include <unordered_map>
+#include <sstream>
+#include <vector>
+#include <memory>
+#include "phonebook.hpp"
+
+namespace ILLIXR {
+
+	/**
+	 * @brief Schema of each record.
+	 *
+	 * name_ should be globally unique.
+	 */
+	class record_header {
+	public:
+		record_header(std::string name_, std::vector<std::pair<std::string, const std::type_info&>> columns_)
+			: id{std::hash<std::string>{}(name_)}
+			, name{name_}
+			, columns{columns_}
+		{ }
+
+		/**
+		 * @brief Compares two schemata.
+		 */
+		bool operator==(const record_header& other) const {
+			// Check pointer first
+			if (this == &other) {
+				return true;
+			}
+
+			if (false
+				|| name != other.name
+				|| columns.size() != other.columns.size()
+				|| id != other.id
+				) {
```

Comment:
Is the `if (false` necessary?

Reply:
No, but it makes the other conditions all start with `|| `.